style_guide.md

```diff
@@ -0,0 +1,286 @@
+
+    - `ccs`
+    - `communal-establishment`
+    - `household`
+    - `individual`
+
+- Blocks from the `individual` folder are also used from the `household` questionnaire.
+
+## Global variables
+
+- Top level arguments
+    
+    - Top-level arguments (`--tla-str`) are used to pass variables that are used in 5 blocks or less.
+    - For each schema created it is possible to pass variables, e.g.: `region_code` and `census_month_year_date`, with different outputs being generated depending on their value.
+    - These are defined in eq-questionnaire-runner, are known at questionnaire launch and won't change during the Census.
+
+- External variables
+    
+    - External variables (`--ext-str`) are used to pass variables that are used in a more than 5 blocks.
+    - External variables are being used. Example of that is `census_month_year_date` and `region_code`.
+    - Example of how we use external variable in `answers`:
+    
+        ```
+        answers: [
+          {
+            id: 'date-of-birth-answer',
+            mandatory: true,
+            type: 'Date',
+            minimum: {
+              value: std.extVar('census_date'),
+              offset_by: {
+                years: -115,
+                months: -2,
+                days: -20,
+              },
+            },
+            maximum: {
+              value: 'now',
+            },
+          },
+        ],
+        ```
+    
+
+- Region Code
+    - Depending on its value, different structure of sections is generated (e.g. certain questions omitted).
+    - Because it's an external variable now, it doesn't have to be passed to blocks and would align with how we use `census_date`.
+    - Example of how we determine if Welsh or English description is used:
+      ```
+      local radioOptions = (
+        if std.extVar('region_code') == 'GB-WLS' then 'Welsh, English, Scottish, Northern Irish or British'
+        else 'Welsh, English, Scottish, Northern Irish or British';
+      );
+      ```
+
+
+## Blocks
+
+- There are two distinct styles we use with block json:
+  - Simple (no variants):
+  
+    ```
+    // Imports
+    
+    {
+      // Block JSON
+    }
+    ```
+  - Variants:
+    ```
+    // Imports
+    
+    // Variant helper methods
+    local questionTitle(variantParam) = (
+      // Variables relevant to this method
+      // Return relevant string or JSON object
+    );
+    
+    // Main variant method - 'question' or 'content'
+    local question(variantParam) = {
+      // Variant JSON
+    };
+    
+    {
+      // Block JSON
+    }
+    ```
+### Variants
+
+- Any time we want to vary the value of a property in a variant of question or content, a method needs to be defined that returns the appropriate variant.
+- Currently for variants we use mainly `isProxy` variable but also job and age related ones: `isEmployed`, `over16`, `over19`, `under19`.
+- Example of question(isProxy=true) and question(isProxy=false) variants:
```

Comment:
I think we should combine the examples into two clear examples. One with just `isProxy` and another with `isProxy` and `isEmployed`. Each example should be under a heading, possibly "Single value variants example" and "Multiple value variants example". Each heading should contain a set of bullet points explaining the key style points and the example Jsonnet.

Reply:
Moved examples under their respective headings, added one more example of a local question for multiple value variants, added bullet points. Thanks!